core/services/headtracker/head_listener.go

```diff
@@ -97,44 +80,52 @@ func (hl *HeadListener) ListenForNewHeads(handleNewHead func(ctx context.Context
 		if ctx.Err() != nil {
 			break
 		} else if err != nil {
-			hl.log.Errorw(fmt.Sprintf("Error in new head subscription, unsubscribed: %s", err.Error()), "err", err)
+			hl.logger.Errorw(fmt.Sprintf("Error in new head subscription, unsubscribed: %s", err.Error()), "err", err)
 			continue
 		} else {
 			break
 		}
 	}
 }
 
+func (hl *headListener) ReceivingHeads() bool {
+	return hl.receivingHeads.Load()
+}
+
+func (hl *headListener) Connected() bool {
```

Comment:
Where is this actually called, I wonder if it's even necessary?

Reply:
From head tracker, `Healthy()` method.
When I get to head_tracker refactoring, this may change. For now I scoped my refactoring to the listener only.